### DISCUSSION TITLE
Add the third party "adbx" in android of the pyproject.toml, but it doesn't have the execute permission

### DISCUSSION THREAD

**Discussion by ccinsz** [beeware/briefcase#1359] [Q&A]
Add the third party "adbx" in pyproject.toml, but it doesn't have the execute permission，it raises:/data/user/0/com.example.demo0529/files/chaquopy/AssetFinder/requirements/adbx/bin/platform-tools/adb: can't execute: Permission denied.

And I use chmod to add it execute permission, but it also raises Permission denied. How can I have the execute permission to use it?

**Comment by mhsmith** [comment]
Sorry, bundling executables with an app on recent versions of Android is very difficult, and maybe impossible (https://github.com/chaquo/chaquopy/issues/605).

Anyway, if you mean [this adbx](https://adbx.software.informer.com/), it appears to be a WIndows app for controlling an Android device, so trying to run it on the device itself makes no sense.

**Reply by mhsmith** [reply]
Or if you mean [this adbx](https://github.com/ysfchn/adbx), again that's designed to be used on the development machine and not on Android itself.

What are you ultimately trying to achieve here?

**Reply by ccinsz** [reply]
I want to use adb server in the android app, so the android app can control the adb.

**Reply by freakboy3742** [reply]
What you are describing makes no sense. `adb` is a tool that runs on a development machine when you are developing an app to access an Android device. It's not something you use *on* an android device.

Again - I ask *what are you trying to do?*. "I want to run ADB" is a solution - a solution that won't work, and appears to be grounded in a fundamental misunderstanding of what ADB *is*. What is the actual task that you are trying to perform that you think requires ADB?

**Reply by ccinsz** [reply]
I saw other android app which can run adb server, like libadb.so, and it can control the adb to send some command to the phone.  So I try to see whether it can use in the android app.

**Reply by freakboy3742** [reply]
Again - "some command"... like *what*. What are you trying to *achieve*? Something like "I want to start another app" or "I want to view the apps logs". 

You are fixating on the idea that the solution to your problem is ADB, and as a result you're ignoring the fact that ADB is almost certainly *not* the way to do what you want to do when on a device.

**Reply by ccinsz** [reply]
For example, a command  "adb shell dumpsys meminfo package_name". The android app uses this command to get its meminfo.

**Reply by mhsmith** [reply]
Anything that comes after `adb shell` is just a command line, so the app can execute it directly: 
```
subprocess.run(["dumpsys", "meminfo", "package_name"])
``` 

However, because of Android's security restrictions, many commands that work within adb will not work within an app, and this may be one of them.

This isn't really a question about BeeWare anymore, so if you need more help with it, please use a more general Android programming forum such as StackOverflow.

**Reply by ccinsz** [reply]
Yes, use subprocess.run(["dumpsys", "meminfo", "package_name"]), it will raise the Permission denied.

**Reply by mhsmith** [reply]
Like I said, this isn't a BeeWare issue, so we can't help you any more with it here.

**Reply by ccinsz** [reply]
Can the beeware use libadb.so to call adb of the phone?

**Reply by mhsmith** [reply]
I don't know anything more about this, you'll have to do your own research.
